### PR TITLE
draft version from setuptools scm

### PR DESCRIPTION
**Pull request by shimwell**
for your consideration , careful before merging as this might break everything :-)

This PR attempts to automatically label versions using setuptools scm which should work with github releases.

To get this working I removed hatch but there must be a way of getting them to work together nicely

also added the option to install the packages needed for testing using ```pip install .[tests]```

### CODE REVIEW

File: src/lazyfunc/__init__.py [hunk 1]
```diff
@@ -1 +1,7 @@
+from importlib.metadata import version, PackageNotFoundError
```
Comment (shimwell) [inline src/lazyfunc/__init__.py:1]:
```suggestion
from importlib.metadata import version
```

File: src/lazyfunc/__init__.py [hunk 2]
```diff
@@ -1 +1,7 @@
+from importlib.metadata import version
+
+__version__ = version("lazyfunc")
+
+# __all__ = ["__version__"]
```
Comment (shimwell) [inline src/lazyfunc/__init__.py:5]:
```suggestion
# avoids unused variable warning
__all__ = ["__version__"]
```

File: pyproject.toml [hunk 3]
```diff
@@ -23,29 +23,26 @@ classifiers = [
...
 
+[tool.setuptools_scm]
+write_to = "src/_version.py"
+
 [project.urls]
 Documentation = "https://github.com/josh-read/lazyfunc#readme"
 Issues = "https://github.com/josh-read/lazyfunc/issues"
 Source = "https://github.com/josh-read/lazyfunc"
 
 [tool.hatch.version]
```
Comment (shimwell) [inline pyproject.toml:34]:
```suggestion
```

File: pyproject.toml [hunk 4]
```diff
@@ -23,29 +23,25 @@ classifiers = [
...
+write_to = "src/_version.py"
+
 [project.urls]
 Documentation = "https://github.com/josh-read/lazyfunc#readme"
 Issues = "https://github.com/josh-read/lazyfunc/issues"
 Source = "https://github.com/josh-read/lazyfunc"
 
-[tool.hatch.version]
-path = "src/lazyfunc/__about__.py"
+path = "src/_version.py"
```
Comment (shimwell) [inline pyproject.toml:34]:
```suggestion
```

**Review by josh-read — CHANGES_REQUESTED** [review]
Thanks for adding this! I tested on my local machine and all seems to be working fine.

File: src/lazyfunc/__init__.py [hunk 5]
```diff
@@ -1 +1,8 @@
+from importlib.metadata import version
+
+__version__ = version("lazyfunc")
+
+# avoids unused variable warning
+__all__ = ["__version__"]
+
```
Comment (shimwell) [inline src/lazyfunc/__init__.py:7]:
```suggestion
__all__ = ["__version__"]

```

File: src/lazyfunc/__init__.py (hunk 2 above)
Comment (josh-read) [inline src/lazyfunc/__init__.py]: Something like this:

```suggestion
__version__ = version("lazyfunc")

__all__ = ["__version__", "LazyFunc"]
```

Comment (shimwell) [reply src/lazyfunc/__init__.py]: Nice, I shall commit this one and resolve

File: src/lazyfunc/__init__.py [hunk 6]
```diff
@@ -1 +1,7 @@
+from importlib.metadata import version
+
+__version__ = version("lazyfunc")
+
+__all__ = ["__version__"]
+
 from .lazy_func import LazyFunc
```
Comment (shimwell) [inline src/lazyfunc/__init__.py:7]:
```suggestion
from importlib.metadata import version

from .lazy_func import LazyFunc


__version__ = version("lazyfunc")
__all__ = ["__version__", "LazyFunc"]
```

**Review by josh-read — APPROVED** [review]

**Pull request merged**